src/storage/wal_replayer.cpp

```diff
@@ -10,20 +10,18 @@ using namespace kuzu::common;
 namespace kuzu {
 namespace storage {
 
-WALReplayer::WALReplayer(WAL* wal) : isRecovering{true}, isCheckpoint{true}, wal{wal} {
-    init();
-}
-
+// COMMIT_CHECKPOINT:   isCheckpoint = true,  isRecovering = false
+// ROLLBACK:            isCheckpoint = false, isRecovering = false
+// RECOVERY_CHECKPOINT: isCheckpoint = true,  isRecovering = true
 WALReplayer::WALReplayer(WAL* wal, StorageManager* storageManager, MemoryManager* memoryManager,
-    Catalog* catalog, bool isCheckpoint)
-    : isRecovering{false}, isCheckpoint{isCheckpoint}, storageManager{storageManager},
-      bufferManager{memoryManager->getBufferManager()},
-      memoryManager{memoryManager}, wal{wal}, catalog{catalog} {
+    BufferManager* bufferManager, Catalog* catalog, WALReplayMode replayMode)
```

Comment:
Maybe we should have a separate code path for different cases?
COMMIT_CHECKPOINT
ROLLBACK
RECOVERY_CHECKPOINT

Reply:
No. I don't think checkpoint and rollback go through wal replayer is a good idea. We should de-couple checkpoint and rollback from wal replayer, which should only be used during recovery. I'm currently working a new design, which I will propose soon.